packages/tradershub/src/flows/RealAccountCreation/RealAccountCreation.tsx

```diff
@@ -21,8 +34,19 @@ const FORM_PROGRESS_STEPS: TSteps = ['Account currency', 'Personal details', 'Ad
  */
 const RealAccountCreation = () => {
     const [isConfirmationDialogOpen, setIsConfirmationDialogOpen] = useState(false);
-    const { currentStep } = useRealAccountCreationContext();
+    const { currentStep, setTotalSteps } = useRealAccountCreationContext();
     const { isModalOpen } = useQueryParams();
+    const { isEU } = useRegulationFlags();
+
+    // Get the form steps based on the user's location
+    const FORM_PROGRESS_STEPS = useMemo(() => getFormSteps(isEU), [isEU]);
+
+    // Set the total steps in the progress bar based on the form steps
+    useEffect(() => {
+        if (FORM_PROGRESS_STEPS.length) {
+            return setTotalSteps(FORM_PROGRESS_STEPS.length);
```

Comment:
may I clarify, why do we need return here?

Reply:
it has the same effect either you include if you want to terminate the execution. There was more logic here previously which needed the return. As for now its the same thing.

Reply:
I understand what it does, I just don't understand why do we need it here.
as there is no logic for terminating the execution, it looks just more confusing. I'd suggest remove unnecessary code